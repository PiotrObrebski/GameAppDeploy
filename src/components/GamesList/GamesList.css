.game-list{
  list-style: none;
}

.game-list-item{
  margin: 10px 0;
  font-family: 'Architects Daughter', cursive; 
}

.game-list-item-link{
  padding: 5px 30px;
  border: 1px solid #8ac926;
  border-radius: 100px;
  text-decoration: none;
  color: #8ac926;
  font-size: 3rem;
}

.game-list-item-link:hover{
  color: #231826;
  background-color: #8ac926;
}
