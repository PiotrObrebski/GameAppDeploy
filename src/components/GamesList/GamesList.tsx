import './GamesList.css';

import React, { ReactElement } from 'react';
import { Link } from 'react-router-dom';

export const GamesList = (props:{
  allGames: {
    gamesArray: IGame[]
  }
}): ReactElement => {
  return (
    <ul className="game-list">
      {props.allGames.gamesArray.map(
        (game, id)=>{
          const {name} = game;
          return(
            <li key={id} className="game-list-item">
              <Link className="game-list-item-link"to={`/${name}`}>{name}</Link>
            </li>
          )
        }
      )}
    </ul>
  
  )
}
