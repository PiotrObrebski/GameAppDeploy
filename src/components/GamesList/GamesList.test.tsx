import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import store from '../../redux/store';
import { GamesList } from './GamesList';

describe("GamesList", () => {
  let component: ShallowWrapper<{ any: any }>;
  const props = {
    allGames: {
      gamesArray: [
        {
          id: "id",
          name: "name",
          intro_text: "intro_text",
          farewell_text: "farewell_text",
          time: 0,
        },
      ]
    }
  };

  beforeEach(() => {
    component = shallow(
      <Provider store={store}>
        <GamesList {...props}/>
      </Provider>
    );
  })

  test("Should mount", () => {
    expect(component.length).toBe(1);
  });

  test("should match snapshot", () => {
    const GamesListRender = renderer
      .create(
        <Provider store={store}>
          <Router>
          <GamesList {...props}/>
          </Router>
        </Provider>
      )
      .toJSON();
    expect(GamesListRender).toMatchSnapshot();
  });
});
