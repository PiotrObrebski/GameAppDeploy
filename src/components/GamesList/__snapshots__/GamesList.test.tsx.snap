// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GamesList should match snapshot 1`] = `
<ul
  className="game-list"
>
  <li
    className="game-list-item"
  >
    <a
      className="game-list-item-link"
      href="/name"
      onClick={[Function]}
    >
      name
    </a>
  </li>
</ul>
`;
